controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -84,10 +88,14 @@ func NewHelper(original, modified client.Object, c client.Client, opts ...Helper
 		return nil, errors.Wrap(err, "failed to remove fields merge patch")
 	}
 
+	// TODO: improve this to have somethings similar to allowPath/IgnorePath
+	diff := cmp.Diff(original, modified)
```

Comment:
We might also want to add that we should really only do this on demand, rather than always calculating it. For now this is ok given the alpha state of this feature. Although

Reply:
I have moved diff out of this PR given that I should figure out a way to make cmp consistent with allowedPaths and ignorePaths, otherwise it will provide false positives